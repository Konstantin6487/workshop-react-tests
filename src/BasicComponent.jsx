import React from 'react';
import { Tabs, TabList } from 'react-tabs';

const BasicComponent = ({ renderTabsTitles, renderTabsContent }) => (
  <Tabs>
    <TabList data-test="tabs-container">{renderTabsTitles()}</TabList>
    {renderTabsContent()}
  </Tabs>
);

export default BasicComponent;
