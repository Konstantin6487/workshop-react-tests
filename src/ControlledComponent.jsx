import React, { Component } from 'react';
import {
  Tab,
  Tabs,
  TabList,
  TabPanel,
} from 'react-tabs';

export default class ControlledComponent extends Component {
  state = { tabIndex: 0 };

  handleSelect = (tabIndex) => {
    this.setState({ tabIndex });
  };

  render() {
    const { tabIndex } = this.state;
    return (
      <Tabs selectedIndex={tabIndex} onSelect={this.handleSelect}>
        <TabList>
          <Tab>Title 1</Tab>
          <Tab>Title 2</Tab>
        </TabList>
        <TabPanel />
        <TabPanel />
      </Tabs>
    );
  }
}
