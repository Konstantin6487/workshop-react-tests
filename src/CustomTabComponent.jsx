import React from 'react';

import {
  Tabs,
  TabList,
  Tab,
  TabPanel,
} from 'react-tabs';

const CustomTab = ({ children }) => (
  <Tab>
    <h1>{children}</h1>
  </Tab>
);

CustomTab.tabsRole = 'Tab'; // Required field to use your custom Tab

const CustomTabComponent = () => (
  <Tabs>
    <TabList>
      <CustomTab>Custom Tab 1</CustomTab>
      <CustomTab>Custom Tab 2</CustomTab>
    </TabList>
    <TabPanel>Panel 1</TabPanel>
    <TabPanel>Panel 2</TabPanel>
  </Tabs>
);

export default CustomTabComponent;
