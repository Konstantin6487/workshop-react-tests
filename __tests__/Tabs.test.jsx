import React from 'react';
import renderer from 'react-test-renderer';
import BasicComponent from '../src/BasicComponent';
import ControlledComponent from '../src/ControlledComponent';
import CustomTabComponent from '../src/CustomTabComponent';

describe('react-test-renderer', () => {
  test('<BasicComponent />', () => {
    const component = renderer.create(<BasicComponent />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<ControlledComponent />', () => {
    const component = renderer.create(<ControlledComponent />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('<CustomTabComponent />', () => {
    const component = renderer.create(<CustomTabComponent />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
