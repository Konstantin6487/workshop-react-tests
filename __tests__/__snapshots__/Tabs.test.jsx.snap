// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-test-renderer <BasicComponent /> 1`] = `
<div
  className="react-tabs"
  data-tabs={true}
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <ul
    className="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-1"
      aria-disabled="false"
      aria-selected="true"
      className="react-tabs__tab react-tabs__tab--selected"
      id="react-tabs-0"
      role="tab"
      tabIndex="0"
    >
      Title 1
    </li>
    <li
      aria-controls="react-tabs-3"
      aria-disabled="false"
      aria-selected="false"
      className="react-tabs__tab"
      id="react-tabs-2"
      role="tab"
      tabIndex={null}
    >
      Title 2
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-0"
    className="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-1"
    role="tabpanel"
  >
    <h2>
      Any content 1
    </h2>
  </div>
  <div
    aria-labelledby="react-tabs-2"
    className="react-tabs__tab-panel"
    id="react-tabs-3"
    role="tabpanel"
  />
</div>
`;

exports[`react-test-renderer <ControlledComponent /> 1`] = `
<div
  className="react-tabs"
  data-tabs={true}
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <ul
    className="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-5"
      aria-disabled="false"
      aria-selected="true"
      className="react-tabs__tab react-tabs__tab--selected"
      id="react-tabs-4"
      role="tab"
      tabIndex="0"
    >
      Title 1
    </li>
    <li
      aria-controls="react-tabs-7"
      aria-disabled="false"
      aria-selected="false"
      className="react-tabs__tab"
      id="react-tabs-6"
      role="tab"
      tabIndex={null}
    >
      Title 2
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-4"
    className="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-5"
    role="tabpanel"
  />
  <div
    aria-labelledby="react-tabs-6"
    className="react-tabs__tab-panel"
    id="react-tabs-7"
    role="tabpanel"
  />
</div>
`;

exports[`react-test-renderer <CustomTabComponent /> 1`] = `
<div
  className="react-tabs"
  data-tabs={true}
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <ul
    className="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-controls={null}
      aria-disabled="false"
      aria-selected="false"
      className="react-tabs__tab"
      id={null}
      role="tab"
      tabIndex={null}
    >
      <h1>
        Custom Tab 1
      </h1>
    </li>
    <li
      aria-controls={null}
      aria-disabled="false"
      aria-selected="false"
      className="react-tabs__tab"
      id={null}
      role="tab"
      tabIndex={null}
    >
      <h1>
        Custom Tab 2
      </h1>
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-8"
    className="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-9"
    role="tabpanel"
  >
    Panel 1
  </div>
  <div
    aria-labelledby="react-tabs-10"
    className="react-tabs__tab-panel"
    id="react-tabs-11"
    role="tabpanel"
  />
</div>
`;
